### ISSUE TITLE
Automatic Releases

### ISSUE THREAD

**Issue by AsherGlick** [AsherGlick/Burrito#333]
As discussed in #328 we will be making a new set of automatic release tools to allow us to release code more frequently and with less effort. 
Testing this release process is going to be quite tricky because the workflow will technically only run on the primary branch. Forks can run the workflows on their primary branches as tests but that is about it. Due to that the tasks have been broken up into smaller chunks that should allow for real-world testing of the build pipelines without needing to test all the features at once..

- [ ] "Weekly Release" Initial Workflow, without actually making a new release
- [ ] Automatic version tags for the weekly release
- [ ] Skip Weekly Release if there are no new commits
- [ ] Add release creation to the weekly release workflow
- [ ] Skip non user facing weekly releases
- [ ] Automatic Descriptions for each weekly release
- [ ] "Major Release" workflow that is run manually
- [ ] "Bugfix Release" workflow that is run manually

**Comment by Masgalor** [comment]
Do we agree, that we bring this topic forward in #343 to the point where we are happy with that PR as a whole, so I can create sub-PRs based on that for every single issue / test-step?

That way we would merge the smaller PRs one by one to finally merge #343 itself .

**Comment by AsherGlick** [comment]
Yes, that sounds good to me. Thanks for being patient with this last bit of that PR, I probably wont be able to sit down and figure it out until at least this weekend given my current schedule.
